File: PROJETO_5 - WESLEY SANTOS/src/Professor.py
class Usuario:
    def __init__(self, nome='', email='', tipo=''):
        self.nome = nome
        self.email = email
        self.tipo = tipo


class Professor(Usuario):
    def __init__(self, nome='', email='', area_de_atuacao=''):
        super().__init__(nome, email, tipo='Professor')
        self.area_de_atuacao = area_de_atuacao
        self.disciplinas_ministradas = []

    def gerar_relatorio(self):
        print(f"Nome: {self.nome}")
        print(f"Área de Atuação: {self.area_de_atuacao}")

    def adicionar_disciplina(self, disciplina):
        self.disciplinas_ministradas.append(disciplina)

File: Projeto_6/Atv-POO6/Projeto_6/Professor.cpp
#include "Professor.h"
#include <iostream>

Professor::Professor(const std::string& nome, const std::string& email, const std::string& areaDeAtuacao)
    : Usuario(nome, email, "Professor"), areaDeAtuacao(areaDeAtuacao) {}

void Professor::adicionarDisciplina(const std::string& disciplina) {
    disciplinasMinistradas.push_back(disciplina);
}

void Professor::gerarRelatorio() const {
    std::cout << "Relatório do Professor:\n";
    std::cout << "Nome: " << nome << "\nEmail: " << email << "\n";
    std::cout << "Área de Atuação: " << areaDeAtuacao << "\nDisciplinas Ministradas:\n";
    for (const auto& disciplina : disciplinasMinistradas) {
        std::cout << "- " << disciplina << "\n";
    }
}
